### PR TITLE
Build docker images via Github actions

### PR DESCRIPTION
**Pull request by codedge**
This PR is going to introduce docker images build via Github actions. It adds a separate _Makefile_ which takes care of the different images, the building and tagging.

Whenever a new release is created the docker images are going to be built from `master` branch.

**Preparation**

Two secretes are needed to login to Docker Hub:
* `DOCKERHUB_USERNAME`
* `DOCKERHUB_PASSWORD`

These need to set up in the repository before.

### CODE REVIEW

**Comment by hillelcoren** [comment]
Nice!!

File: README.md [hunk 1]
```diff
@@ -161,8 +154,3 @@ environment:
 to your `docker-compose.yml`.
 
 This generated log file will only hold Invoice Ninja information.
-
-
-### Known issues
-
-Phantomjs doesn't work on linux alpine https://github.com/ariya/phantomjs/issues/14186.
```
Comment (lalop) [inline README.md:168]: why did you remove that ?

File: alpine/Dockerfile [hunk 2]
```diff
@@ -1,69 +1,85 @@
...
-RUN docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ \
-    && docker-php-ext-configure gmp \
-    && docker-php-ext-install iconv mbstring pdo pdo_mysql zip gd gmp opcache \
-    && echo "php_admin_value[error_reporting] = E_ALL & ~E_NOTICE & ~E_WARNING & ~E_STRICT & ~E_DEPRECATED">>/usr/local/etc/php-fpm.d/www.conf
+WORKDIR /var/www/app
+
+COPY ./alpine/entrypoint.sh /usr/local/bin/docker-entrypoint
+RUN chmod +x /usr/local/bin/docker-entrypoint
+
+ENV PHANTOMJS="phantomjs-2.1.1-linux-x86_64"
```
Comment (lalop) [inline alpine/Dockerfile:16]: It doesn't seem to be used actually, we may remove this line

File: alpine/entrypoint.sh [hunk 3]
```diff
@@ -1,47 +1,44 @@
...
-		fi
-	done
-fi
 
-# compare public volume version with image version
-if [ ! -e /var/www/app/public/version ] || [ "$INVOICENINJA_VERSION" != "$(cat /var/www/app/public/version)" ]; then
-  echo 'clone public directory'
-  cp -Rp /var/www/app/docker-backup-public/* /var/www/app/public/
-  echo $INVOICENINJA_VERSION > /var/www/app/public/version
+if [ "$1" = 'php-fpm' ] || [ "$1" = 'bin/console' ]; then
```
Comment (lalop) [inline alpine/entrypoint.sh:10]: why this condition ?

File: config/php/php-cli.ini [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+opcache.enable_cli=1
+opcache.fast_shutdown=1
+opcache.memory_consumption=128
+opcache.interned_strings_buffer=8
+opcache.max_accelerated_files=4000
+opcache.revalidate_freq=60
+# http://symfony.com/doc/current/performance.html
+realpath_cache_size = 4096K
+realpath_cache_ttl = 600
```
Comment (lalop) [inline config/php/php-cli.ini:14]: php.ini can be load in cli context, that way you can avoid duplication.

File: debian/Dockerfile [hunk 5]
```diff
@@ -1,11 +1,16 @@
-ARG PHP_IMAGE_TAG=7.2-fpm-stretch
-FROM php:${PHP_IMAGE_TAG}
+ARG PHP_VERSION=7.2
+ARG INVOICENINJA_VERSION
+
+FROM php:${PHP_VERSION}-fpm-stretch
 
 LABEL maintainer="Samuel Laulhau <sam@lalop.co>"
```
Comment (lalop) [inline debian/Dockerfile:6]: you can add your name here too ;)

File: debian/Dockerfile [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
 
 #####
 # SYSTEM REQUIREMENT
 #####
+COPY ./alpine/entrypoint.sh /usr/local/bin/docker-entrypoint
```
Comment (lalop) [inline debian/Dockerfile:11]: is that normal you copy the alpine entrypoint ?

Comment (codedge) [reply debian/Dockerfile]: Good catch. Fixed.

File: debian/Dockerfile (hunk 5 above)
Comment (codedge) [reply debian/Dockerfile]: Fixed 😉

File: config/php/php-cli.ini (hunk 4 above)
Comment (codedge) [reply config/php/php-cli.ini:14]: I saw this separation in other projects... in cli context the memory limit is set to `2G`. See line 16 in that file:
```php
memory_limit = 2G
```
So only on cli we have a different setting for `memory_limit`. For the normal usage we did not explicitly set something which then uses the default of `128M`.

File: alpine/Dockerfile (hunk 2 above)
Comment (codedge) [reply alpine/Dockerfile]: Fixed

File: README.md (hunk 1 above)
Comment (codedge) [reply README.md:168]: I thought as we use phantomJS in alpine, see line 
```php
RUN cd /usr/share \
    && curl -s -L https://github.com/Overbryd/docker-phantomjs-alpine/releases/download/2.11/phantomjs-alpine-x86_64.tar.bz2 | tar xj \
    && ln -s /usr/share/phantomjs/phantomjs /usr/local/bin/phantomjs
```

it works - or doesn't it?

File: alpine/entrypoint.sh (hunk 3 above)
Comment (codedge) [reply alpine/entrypoint.sh]: Fixed. Some old check.

File: README.md (hunk 1 above)
Comment (lalop) [reply README.md:168]: not last time I tried

Comment (codedge) [reply README.md:168]: As agreed, the local PhantomJS version is removed from `alpine` version. The cloud version is the preferred one.
